### ISSUE TITLE
Libftl.ddl crash

### ISSUE THREAD

**Issue by ZoeBell** [WatchBeam/obs-studio-ftl#31]
I posted this to the OBS forums and they redirected me: Please let me know if I need to post this somewhere else, I apologize.

I just started setting up OBS-Studio tonight to start streaming on Beam(was using classic OBS to Twitch stream and record videos previously), got my overlay looking nice using Streamjar, got my sources and settings sorted on OBS 0.16.4-ftl.15 64bit and can't actually stream because OBS crashes every time I try to start streaming. Whether I "start streaming" using the button on OBS or the hotkeys I set up.

I've noted a portion of my most recent crash log below. Both with my use of version 15 and 14. Any help would be great! Thank you in advance. 

https://gist.github.com/e6b537d2285e249eca53300d7ab4d54b

Unhandled exception: c0000005
Fault address: 7FEECE329EF (c:\program files (x86)\obs studio - ftl\obs-plugins\64bit\libftl.dll)
libobs version: 0.16.2-ftl.14.pre
Windows version: 6.1 build 7601 (revision 23543)
CPU: Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz

Unhandled exception: c0000005
Date/Time: 2016-11-05, 18:58:53
Fault address: 7FEECE329EF (c:\program files (x86)\obs studio - ftl\obs-plugins\64bit\libftl.dll)
libobs version: 0.16.4-ftl.15
Windows version: 6.1 build 7601 (revision 23543)
CPU: Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz

**Comment by JimmyMac3** [comment]
In the Audio settings SAMPLE RATE must be set to 48Khz
In the Output settings ENCODER must be set to x264

Any other options for those 2 settings will cause OBS to crash. FTL is not compatible with 44.1Khz or any other encoder at the moment

**Comment by ZoeBell** [comment]
I set it up initially with all the recommended settings noted in the change logs. So 48Khz(which was default) and x264 Encoder settings. I did, afterwards, try to mess with ALL the settings but the crashes were happening regardless with the libobs. Current settings are run exactly as the change log suggests, and still, same crashes I had originally.

**Comment by sslivins** [comment]
@ZoeBell thanks for the logs, a crash wouldnt be caused by the wrong audio rate or any settings in general....let me see what i can figure out based on the information you provided.  also what version of windows are you on?  windows 7?

**Comment by ZoeBell** [comment]
That's correct. Windows 7 64bit. I can record videos just fine, but crash when I attempt to stream. If more information is needed I can send it once I'm off work today. Thank you!

**Comment by sslivins** [comment]
are you trying to record and stream at the same time?  i dont believe this would cause a crash though but that currently is not supported with FTL

**Comment by ZoeBell** [comment]
I'm not trying to record at the same time. I don't have that option selected.

I was recording videos to test sound and cam levels but would stop the recording, view the video, delete it, and then attempt to start streaming. Then I would crash. Here's a few of the reports from version 14 and 15. 

https://obsproject.com/forum/attachments/crash-2016-11-05-18-10-01-txt.20851/
https://obsproject.com/forum/attachments/crash-2016-11-05-19-21-17-txt.20850/
https://obsproject.com/forum/attachments/crash-2016-11-05-18-58-53-txt.20849/
https://obsproject.com/forum/attachments/crash-2016-11-05-19-29-01-txt.20848/

Hopefully those are viewable. 

Thanks!

**Comment by sslivins** [comment]
can you post the main logs also, they are found in %appdata%\obs-studio\logs

just post 2 of the most recent ones that exhibited the problem (i assume it always happens so just the last 2)

**Comment by ZoeBell** [comment]
[2016-11-05 18-56-20.txt](https://github.com/WatchBeam/obs-studio-ftl/files/577179/2016-11-05.18-56-20.txt)
[2016-11-05 18-58-45.txt](https://github.com/WatchBeam/obs-studio-ftl/files/577180/2016-11-05.18-58-45.txt)

**Comment by ZoeBell** [comment]
If it helps, I just started it up again so here would be the absolute latest logs. 
[Crash 2016-11-07 18-38-29.txt](https://github.com/WatchBeam/obs-studio-ftl/files/577190/Crash.2016-11-07.18-38-29.txt)
[2016-11-07 18-37-05.txt](https://github.com/WatchBeam/obs-studio-ftl/files/577191/2016-11-07.18-37-05.txt)

**Comment by sslivins** [comment]
so i know what caused the crash (fix will be in next version) which is the dns lookup for ingest-sjc.beam.pro failed, but as for why that would fail im a bit confused still.  are you able to ping ingest-sjc.beam.pro?

**Comment by ZoeBell** [comment]
![ping test](https://cloud.githubusercontent.com/assets/23288616/20089353/3d973320-a53b-11e6-97e0-ca145c028949.PNG)
No issue connecting. D: 
Glad you were able to see what was causing the crash though!

**Comment by sslivins** [comment]
Can you try using 'Auto' as your ingest rather than sjc and see if that works

**Comment by ZoeBell** [comment]
Sorry, election got me distracted from streaming or...anything else. 

Do you mean to try a PING test using auto instead of sjc? That one also came back as 0% loss. Not sure if there was something else meant by that.

**Comment by sslivins** [comment]
no, i meant try connecting to the "Auto" ingest rather than San Jose directly.  Im not sure i expect a different result but i'm curious to see.  Also trying to change to Seattle, do you experience the same behavior.  I would like to get to the bottom of this, i suspect it's related to unsupported API's in windows 7 but i want to be sure.  I will likely create a custom build for you to try out with more debugging information but probably not until next week.  Thanks again for helping debug this, I know it's a bit tedious but it's appreciated

**Comment by ZoeBell** [comment]
Okay, sorry about that misunderstanding. I changed it to Auto and still crashed, just the same as it had been. I see there's a new update so I can attempt to install that one this weekend? I probably won't be able to update anyways until Saturday but let me know if I can help!

**Comment by sslivins** [comment]
let me get you a new version later in the week that has more logging info so i can see why it's failing to find any ingests.  i'll ping you when it's ready.  thanks again for your help

**Comment by ZoeBell** [comment]
So, I finally got around to downloading 0.16.4-ftl.15 again and it's suddenly working. Perfectly. I didn't stream for long, just testing. But, no initial crash! Let me know if there's any logs you'd like if it helps in the future! Otherwise, looks good?

**Comment by sslivins** [comment]
sure, post them (logs->upload current) after a successful stream (the duration doesnt matter)

**Comment by ZoeBell** [comment]
[2016-11-15 19-34-31.txt](https://github.com/WatchBeam/obs-studio-ftl/files/593704/2016-11-15.19-34-31.txt)

Just did a quick test stream, here it is!

**Comment by sslivins** [comment]
looks like you are using rtmp in that last log...can you try using the 'ftl beam service' again?

**Comment by ZoeBell** [comment]
Well, that's my fault for not checking the settings after I downloaded the latest version. Whoops. It is crashing in FTL per the usual. 
[Crash 2016-11-16 18-14-42.txt](https://github.com/WatchBeam/obs-studio-ftl/files/596386/Crash.2016-11-16.18-14-42.txt)
[2016-11-16 18-13-00.txt](https://github.com/WatchBeam/obs-studio-ftl/files/596388/2016-11-16.18-13-00.txt)

**Comment by sslivins** [comment]
@ZoeBell try this build and send me the logs (it shouldnt crash but I expect things to still fail)  https://dl.dropboxusercontent.com/u/20701844/obs-studio-0.16.5-ftl.zoe.msi

**Comment by ZoeBell** [comment]
[2016-11-19 11-56-10.txt](https://github.com/WatchBeam/obs-studio-ftl/files/601694/2016-11-19.11-56-10.txt)
No crash, just the error message as you suggested. Here are the logs!

**Comment by sslivins** [comment]
thanks @ZoeBell this helped a lot.  can you try this as a possible fix https://dl.dropboxusercontent.com/u/20701844/obs-studio-0.16.5-ftl.16.ssl.msi

**Comment by sslivins** [comment]
@ZoeBell i believe i have found the root cause which related to features that are unsupported in windows 7.  The latest version I posted should resolve this issue, please let me know if you continue to have problems:  https://github.com/WatchBeam/obs-studio-ftl/releases/tag/ftl.17

**Comment by ZoeBell** [comment]
Okay, great! I feel dumb asking this but how do I install that? Am I overwriting or unpacking...or am I missing the .exe?

**Comment by sslivins** [comment]
sorry, that was another issue which is fixed now, just go back to the release and redownload and you should be good to go

**Comment by ZoeBell** [comment]
[2016-11-20 22-20-06.txt](https://github.com/WatchBeam/obs-studio-ftl/files/602835/2016-11-20.22-20-06.txt)

[It's Alive!](https://cdn.meme.am/instances/250x250/60303857.jpg)

**Comment by ZoeBell** [comment]
Thank you so much! I'm stoked I can get in some streaming now before the holidays haha WOO! Hopefully this will helps others too.

:100: :

**Comment by sslivins** [comment]
no, thank you!  It definitely helps others, i've been getting more and more reports of this problem so it's good to get it fixed
